Khan/wonder-blocks
# Migrate CheckboxGroup to Storybook

## Summary:
Migrate CheckboxGroup to Storybook

A lot of the documentation was already there in styleguidist,
so I mostly used that and built on top of it a little.

Issue: https://khanacademy.atlassian.net/browse/WB-1292

<img width="1138" alt="Screen Shot 2022-06-13 at 5 56 07 PM" src="https://user-images.githubusercontent.com/13231763/173471241-dd4b3789-bd5d-48a1-8125-1f941a88aea6.png">
<img width="1045" alt="Screen Shot 2022-06-13 at 5 56 18 PM" src="https://user-images.githubusercontent.com/13231763/173471254-2d23a3ec-ea5d-4f5e-b3a0-f205e345b96e.png">
<img width="1042" alt="Screen Shot 2022-06-13 at 5 56 24 PM" src="https://user-images.githubusercontent.com/13231763/173471262-89709d6b-7865-40bb-bc69-2dc9306be41d.png">
<img width="1074" alt="Screen Shot 2022-06-13 at 5 56 32 PM" src="https://user-images.githubusercontent.com/13231763/173471269-0b8eb43c-139b-4e24-95b9-af61bc592771.png">
<img width="1081" alt="Screen Shot 2022-06-13 at 5 56 35 PM" src="https://user-images.githubusercontent.com/13231763/173471279-6431a660-63f0-4bb7-994e-55842428809a.png">


## Test plan:
- Go to http://localhost:6061/?path=/docs/form-checkboxgroup--default
- Confirm that the syntax hilighting in the usage looks correct
- Confirm that controls alter the Default story as expected
- Make sure the story descriptions make sense and there are no typos
